src/rpc/state_api.rs

```diff
@@ -377,3 +377,27 @@ pub(in crate::rpc) async fn state_get_randomness_from_beacon<
     )?;
     Ok(LotusJson(value.to_vec()))
 }
+
+/// Get read state
+pub(in crate::rpc) async fn state_read_state<DB: Blockstore + Send + Sync + 'static>(
+    data: Data<RPCState<DB>>,
+    Params(LotusJson((addr, tsk))): Params<LotusJson<(Address, TipsetKeys)>>,
+) -> Result<LotusJson<ApiActorState>, JsonRpcError> {
+    let ts = data.chain_store.load_required_tipset(&tsk)?;
+    let actor = data
+        .state_manager
+        .get_actor(&addr, *ts.parent_state())?
+        .ok_or("Actor address could not be resolved")?;
+    let blk = data
+        .state_manager
+        .blockstore()
+        .get(&actor.state)?
+        .ok_or("Failed to get block from blockstore")?;
+    let state = fvm_ipld_encoding::from_slice::<Vec<Cid>>(&blk)?[0];
```

Comment:
nit: All of this could be a method in `state_manager`, similar to other RPC methods

Reply:
@ruseinov I think the best place for `<category>_<endpoint>` is in `src/rpc/<category>.rs`. Do you have something else in mind?

Reply:
And then that method is used the way @elmattic suggests:

https://github.com/ChainSafe/forest/blob/cc652738faa0c8e7d626d474e8e3838912ce0dcc/src/rpc/state_api.rs#L100-L111